R/get-osmdata.R

```diff
@@ -256,6 +303,34 @@ get_metadata <- function (obj, doc) {
     return (obj)
 }
 
+is_datetime <- function (x) {
```

Comment:
This whole function can be written as something like:

``` r
is_datetime <- function (x) {
    ptn <- "[0-9]{4}(\\-|\\/)[0-9]{2}(\\-|\\/)[0-9]{2}\\s*[0-9]{1,2}\\:[0-9]{1,2}\\:[0-9]{1,2}"
    grepl (ptn, x)
}
```
That is maybe a bit of overkill, as the timestamps passed to this are only those generated by overpass, right? In which case they are more standardised that that assumes, so could be simplified to:
``` r
    ptn <- "[0-9]{4}\\-[0-9]{2}\\-[0-9]{2}\\s*[0-9]{2}\\:[0-9]{2}\\:[0-9]{2}"
```

Reply:
You are right, the inputs should come from an `opq` object, thus, the datetimes already have passed `check_datetime`. I modified a bit the regex to pass the tests (no need to escape - nor :):

```r
ptn <- "[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}[A-Z]"
```

I don't get what ptn stand for, but it will work.